models.py

```diff
@@ -146,3 +251,36 @@ def callbacks_unet(callbacks_config):
         callbacks=[experiment_timing, training_monitor, validation_monitor,
                    model_checkpoints, lr_scheduler, early_stopping, neptune_monitor,
                    ])
+
+
+def multiclass_weighted_segmentation_loss(output, target, w0, sigma):
+    '''
+    w1 is temporarily torch.ones - it should handle class imbalance for thw hole dataset
+    '''
+    distance = target[:, 1, :, :]
+    target = target[:, 0, :, :].long()
+    w1 = Variable(torch.ones(distance.size()), requires_grad=False)  # TODO: fix it to handle class imbalance
+    if torch.cuda.is_available():
+        w1 = w1.cuda()
+    weights = get_weights(distance, w1, w0, sigma)
+    torch_softmax = torch.nn.Softmax2d()
+    probabilities = torch_softmax(output)
+    negative_log_likelihood = torch.nn.NLLLoss2d(reduce=False)
+    loss_per_pixel = negative_log_likelihood(torch.log(probabilities), target)
+    loss = torch.mean(loss_per_pixel * weights)
+    return loss
+
+
+def get_weights(d, w1, w0, sigma):
```

Comment:
@apyskir this is private function so I would go with _get_weights I think

Reply:
@jakubczakon OK